go/mysql/datetime/timeparts.go

```diff
@@ -70,7 +71,16 @@ func (tp *timeparts) toDateTime() (DateTime, bool) {
 			second:     uint8(tp.sec),
 			nanosecond: uint32(tp.nsec),
 		},
-	}, true
+	}
+
+	l := prec
+	if prec < 0 {
```

Comment:
Is it worth explicitly checking for -1 in case e.g. -2 might mean something else in the future? Not sure... just came to mind.

Reply:
Right, but in that case things don't work anywhere so not sure how that helps. It trades one undefined behavior for another and I don't really see a strong preference for either anything way.